modules/tide_content_collection/src/Plugin/Field/FieldWidget/ContentCollectionConfigurationWidget.php

```diff
@@ -1652,73 +1743,129 @@ public function massageFormValues(array $values, array $form, FormStateInterface
-        ;
-      }
-      else {
-        $config['interface']['keyword']['fields'] = ['title'];
+      if (!empty($internal_sort)) {
+        $config['internal']['sort'][] = $internal_sort;
       }
 
-      if (!empty($value['tabs']['filters']['interface_filters']['advanced_filters'])) {
-        $advanced_filters = array_keys($value['tabs']['filters']['interface_filters']['advanced_filters']);
-        if (!empty($advanced_filters)) {
-          foreach ($advanced_filters as $field_id) {
-            $referenced_field = $this->indexHelper->getEntityReferenceFieldInfo($this->index, $field_id);
-            if (!empty($value['tabs']['filters']['interface_filters']['advanced_filters'][$field_id]['allow']) && $value['tabs']['filters']['interface_filters']['advanced_filters'][$field_id]['allow']) {
-              $field = [];
-              $field['elasticsearch-field'] = $field_id;
-              $field['options']['label'] = $value['tabs']['filters']['interface_filters']['advanced_filters'][$field_id][$field_id . '_label'] ?? '';
-              $field['options']['placeholder'] = $value['tabs']['filters']['interface_filters']['advanced_filters'][$field_id][$field_id . '_placeholder'] ?? '';
-              if (!empty($value['tabs']['filters']['interface_filters']['advanced_filters'][$field_id][$field_id . '_options'])) {
-                foreach ($value['tabs']['filters']['interface_filters']['advanced_filters'][$field_id][$field_id . '_options'] as $index => $reference) {
-                  if (!empty($reference['target_id'])) {
-                    $target_id = (int) $reference['target_id'];
-                    $entity = $this->entityTypeManager->getStorage($referenced_field['target_type'])->load($target_id);
-                    if (!empty($entity)) {
-                      $field['options']['values'][] = [$target_id => $entity->label()];
+      // Filters Layout.
+      if (!empty($value['tabs']['filters']['show_interface_filters']) && $value['tabs']['filters']['show_interface_filters']) {
+        if (!empty($value['tabs']['filters']['interface_filters']['keyword']['allow_keyword_search']) && $value['tabs']['filters']['interface_filters']['keyword']['allow_keyword_search']) {
+          // Required field Type.
+          $config['interface']['keyword']['type'] = 'basic';
+          $config['interface']['keyword']['label'] = $value['tabs']['filters']['interface_filters']['keyword']['label'] ?? '';
+          $config['interface']['keyword']['placeholder'] = $value['tabs']['filters']['interface_filters']['keyword']['placeholder'] ?? '';
+          if (!empty($settings['filters']['enable_keyword_selection']) && $settings['filters']['enable_keyword_selection']) {
+            $config['interface']['keyword']['fields'] = $value['tabs']['filters']['interface_filters']['keyword']['fields'] ? array_values(array_filter($value['tabs']['filters']['interface_filters']['keyword']['fields'])) : [];
+            ;
```

Comment:
@joshua-salsadigital what is this?

Reply:
Retrieve only the enabled values from checkboxes.

Reply:
@joshua-salsadigital what does the single semi-colon do on this line?